pkg/controllers/management/restrictedadminrbac/fleet_test.go

```diff
@@ -0,0 +1,277 @@
+					GetFunc: func(namespace string, name string) (*k8srbac.RoleBinding, error) {
+						return &k8srbac.RoleBinding{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:            name,
+								Namespace:       namespace,
+								Labels:          map[string]string{rbac.RestrictedAdminClusterRoleBinding: "true"},
+								OwnerReferences: ownerRefs,
+							},
+							RoleRef:  roleRef,
+							Subjects: []k8srbac.Subject{},
+						}, nil
+					},
+				}
+
+				c.mockRBInterface = &fakes.RoleBindingInterfaceMock{
+					UpdateFunc: func(rb *k8srbac.RoleBinding) (*k8srbac.RoleBinding, error) {
+						assert.Equal(t, rb, expected)
+						return expected, nil
+					},
+				}
+			},
+			wantErr: false,
+		},
+		{
+			name: "one previously existing correct rolebinding",
+			setup: func(c *mockController) {
+				c.mockRBLister = &fakes.RoleBindingListerMock{
+					GetFunc: func(namespace string, name string) (*k8srbac.RoleBinding, error) {
+						return expected, nil
+					},
+				}
+			},
+			wantErr: false,
+		},
+		{
+			name: "unexpected error in Get call",
+			setup: func(c *mockController) {
+				c.mockRBLister = &fakes.RoleBindingListerMock{
+					GetFunc: func(namespace string, name string) (*k8srbac.RoleBinding, error) {
+						return nil, errors.New("Unexpected error ABC")
+					},
+				}
+			},
+			wantErr:     true,
+			expectedErr: "Unexpected error ABC",
+		},
+		{
+			name: "unexpected error in Create call",
+			setup: func(c *mockController) {
+				c.mockRBLister = &fakes.RoleBindingListerMock{
+					GetFunc: func(namespace string, name string) (*k8srbac.RoleBinding, error) {
+						return nil, &k8serrors.StatusError{ErrStatus: metav1.Status{Reason: metav1.StatusReasonNotFound}}
+					},
+				}
+
+				c.mockRBInterface = &fakes.RoleBindingInterfaceMock{
+					CreateFunc: func(rb *k8srbac.RoleBinding) (*k8srbac.RoleBinding, error) {
+						return nil, errors.New("Unexpected error ABC")
+					},
+				}
+			},
+			wantErr:     true,
+			expectedErr: "Unexpected error ABC",
+		},
+		{
+			name: "unexpected error in Update call",
+			setup: func(c *mockController) {
+				c.mockRBLister = &fakes.RoleBindingListerMock{
+					GetFunc: func(namespace string, name string) (*k8srbac.RoleBinding, error) {
+						return &k8srbac.RoleBinding{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:            name,
+								Namespace:       namespace,
+								Labels:          map[string]string{},
+								OwnerReferences: ownerRefs,
+							},
+							RoleRef:  roleRef,
+							Subjects: subjects,
+						}, nil
+					},
+				}
+
+				c.mockRBInterface = &fakes.RoleBindingInterfaceMock{
+					UpdateFunc: func(rb *k8srbac.RoleBinding) (*k8srbac.RoleBinding, error) {
+						return nil, errors.New("Unexpected error ABC")
+					},
+				}
+			},
+			wantErr:     true,
+			expectedErr: "Unexpected error ABC",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockCtrl := newMockController(t)
```

Comment:
Nit: You don't have to change this, but generally we like to have these tests run in parallel. The actual difference is not large, so you don't need to do it, but it's definitely there.
```suggestion
	for _, tt := range tests {
	        tt := tt
		t.Run(tt.name, func(t *testing.T) {
		        t.Parallel()
			mockCtrl := newMockController(t)
```

Reply:
Another small thing learned today, excellent! Addressed as 46985b92b